ext/scheduler/airflow2/compiler/compiler.go

```diff
@@ -48,6 +48,11 @@ func (com *Compiler) Compile(schedulerTemplate []byte, namespaceSpec models.Name
 		}
 	}
 
+	var metadata *models.JobSpecMetadata
+	if jobSpec.Metadata != (models.JobSpecMetadata{}) {
+		metadata = &jobSpec.Metadata
+	}
```

Comment:
what is this check exactly? Are we checking for `nil`?

Reply:
The reason I use the address instead of the values is to ensure that if it's empty, it won't get compiled at all in the template. The check in line 52 is basically checking if the `jobSpec.Metadata` is not default value (meaning it's not empty), then I will assign the address to the `metadata` shown by line 53.

Reply:
Oh, but what if metadata is not empty but it doesn't have resources configs?

Reply:
Well, good point. It should've been for the resource only. I'll update it.

Reply:
It's updated